js/main.js

```diff
@@ -1,45 +1,491 @@
+  if (isNaN(columns)) {
+    columns = rows;
+    rows = 0;
+  }
+  let shipCanBePlaced;
+  shipCanBePlaced = canBePlaced(
+    PLAYER_BOARD,
+    rows,
+    columns,
+    verticalCheckbox === true ? "vertical" : "horizontal",
+    Object.values(playerShips)[shipCounter]
+  );
+  if (shipCanBePlaced === true) {
+    placeShips(
+      PLAYER_BOARD,
+      rows,
+      columns,
+      verticalCheckbox === true ? "vertical" : "horizontal",
+      Object.values(playerShips)[shipCounter]
+    );
+    shipCounter++;
+  }
+  activeTurn();
+}
+
+//Rendering the ships on the game board
+function renderShips(board) {
+  let shipCanBePlaced;
+  let rows;
+  let columns;
+  let idx = Math.round(Math.random());
+  const vertOrHoriz = ["vertical", "horizontal"];
+
+  Object.values(cpuShips).forEach((shipObj) => {
+    do {
+      rows = getRandomNum(0, HEIGHT);
+      columns = getRandomNum(0, LENGTH);
+      shipCanBePlaced = canBePlaced(
+        board,
+        rows,
+        columns,
+        vertOrHoriz[idx],
+        shipObj
+      );
+    } while (!shipCanBePlaced);
+    placeShips(board, rows, columns, vertOrHoriz[idx], shipObj);
+    idx = Math.round(Math.random());
+  });
+}
+
+//Modifying the header based on the current games state
+function activeTurn() {
+  if (shipCounter <= 6) {
+    activePlayerEl.innerHTML = `Place your ${
+      Object.values(playerShips)[shipCounter].name
+    } ${Object.values(playerShips)[shipCounter].length} Tiles `;
+  } else if (winner !== null) {
+    playAgainBtn.innerHTML = "Play Again";
+    activePlayerEl.innerHTML = `${winner === 1 ? "PLAYER WINS!" : "CPU WINS!"}`;
+    document.body.appendChild(playAgainBtn);
+    return;
+  } else {
+    verticalLabelEl.style.display = "none";
+    activePlayerEl.innerHTML = `${turn === 1 ? "PLAYER'S TURN" : "CPU'S turn"}`;
+  }
+  return sleep(700);
+}
+
+//Checking if the ship can be placed on the selected div
+function canBePlaced(board, rows, columns, vertOrHoriz, ship) {
+  if (vertOrHoriz === "horizontal") {
+    for (let i = columns; i < columns + ship.length; i++) {
+      if (i >= LENGTH || board[rows][i] !== 0) return false;
+    }
+  } else {
+    for (let i = rows; i < rows + ship.length; i++) {
+      if (i >= LENGTH || board[i][columns] !== 0) return false;
+    }
+  }
+  return true;
+}
+
+//Actually placing the ships after confirming that they can be placed
+function placeShips(board, rows, columns, vertOrHoriz, ship) {
+  if (vertOrHoriz === "horizontal") {
+    for (let i = columns; i < columns + ship.length; i++) {
+      board === PLAYER_BOARD
+        ? PLAYER_DIVS[rows * LENGTH + i].classList.add(
+            "player-ship",
+            "horizontal"
+          )
+        : CPU_DIVS[rows * LENGTH + i].classList.add("cpu-ship", "horizontal");
+      board === PLAYER_BOARD
+        ? (PLAYER_DIVS[rows * LENGTH + i].id = `${ship.key}`)
+        : (CPU_DIVS[rows * LENGTH + i].id = `${ship.key}`);
+      ship.firstIndex = `${rows}${columns}`;
```

Comment:
If you're going to add a `firstIndex` property to ships, better add it to the initial object as well, to help with clarity. That way you'll always know what properties are part of the object instead of having to look through the code to determine